rfcs/select-component.md

```diff
@@ -0,0 +1,191 @@
+# Select Component
+
+### Exports
+
+* `StatefulSelect`
+* `StatefulSelectContainer`
+* `Select`
+* `StyledSelectInput`
+* `StyledOption`
+* `StyledDropDown`
+* `StyledLabel`
+* `StyledHint`
+
+### `Select` and `StatefulSelect` API
+
+* `type: TYPE.SEARCH | TYPE.SELECT`:
+  type of component to be in select or search mode (with lookup input)
```

Comment:
What's the downside of always allowing search?

Even for normal HTML `<select>` elements, if you open it and start typing an option prefix, the browser starts selecting items that match. I think if our Select doesn't support that, it's inferior to the native select – so might as well just make every select searchable?

Reply:
Currently we have no such UI design for Search + single, it's always Search+multiple. Search+single will require to change logic of our Input. Otherwise no issues, so anyone can switch on Search mode by allowing this type

Reply:
I think this is search + single, no?

![image](https://user-images.githubusercontent.com/875591/44289789-a1b32b00-a22a-11e8-92e3-5122612520cf.png)

Can we just remove this prop and allow search for all types?

Reply:
this one is not yet single, it can be both. I saw only single pick for Select mode.
We can allow search for all, but as per current idea they are separate with Select
@nguyensomniac correct me if I am wrong